feat(terraform-modules) add a new azure inbound agents modules

dduportal opened the pull request:
This PR comes from the work on https://github.com/jenkins-infra/helpdesk/issues/3954.

This PR is needed by https://github.com/jenkins-infra/azure/pull/693 to set up the NSG rules for the kubernetes agents hosted introduced in the linked PR.

Notes about this new modules:

- It only manages NSG rules for now. It is basically a "set up Azure NSG in a subnet for inbound agents only). 
  - The outbound HKP rules or inbound SSH rules are kept only on the Azure VM module as needed
  - Potential for using this module as part of Azure VM but later (to factorize the "inbound " rules as its repeated on both modules
- The permission part is kubernetes specific and handled by kubernetes management (compared to the Azure RBAC setup on Azure VM agents).
- This module should be reusable for release.ci and infra.ci soon
- This module is thought to be instanciated multiple time for a given controller in the case of different subnets (either different AKS clusters and/or different node pools with distinct subnets).

timja commented on terraform/modules/azure-jenkinsinfra-kubernetes-agents/main.tf:
```diff
@@ -0,0 +1,121 @@
+  source_port_range       = "*"
+  source_address_prefixes = data.azurerm_subnet.kubernetes_agents.address_prefixes
+  destination_port_ranges = [
+    "80",    # HTTP for inbound websocket
```
is this actually needed?

dduportal replied:
I don't think so good catch. I've removed it 👍

dduportal commented:
Updated the PR:

- Naming of locals (pairing code review with @hervelemeur)
- Removed the HTTP/80 port (review by @timja )
- Renamed the module to "inbound": there is nothing specific to kubernetes 
- A bit of fmt
